### PR TITLE
Store state data in zookeeper

### PR DESCRIPTION
**Pull request by NikitaUnisikhin**

### CODE REVIEW

File: ch_tools/chadmin/cli/object_storage_group.py [hunk 1]
```diff
@@ -101,10 +105,10 @@ def object_storage_group(ctx: Context, disk_name: str) -> None:
...
 )
 @option(
-    "--store-state",
-    "store_state",
-    is_flag=True,
-    help=("Write total size of orphaned objects to log file."),
+    "--store-state-path",
+    "store_state_path",
+    default="",
+    help=("Zookeeper node path for storage total size of orphaned objects."),
```
Comment (Alex-Burmak) [inline ch_tools/chadmin/cli/object_storage_group.py:111]: I think we need both options - store state in local file and store state in ZooKeeper. There are ClickHouse clusters with and without ZooKeeper.

File: ch_tools/monrun_checks/ch_orphaned_objects.py [hunk 2]
```diff
@@ -4,12 +4,24 @@
...
+    "--store-state-local",
+    "store_state_local",
+    is_flag=True,
+    help="Get total size of orphaned objects from local file.",
+)
+@click.option(
+    "--store-state-zk-path",
+    "store_state_zk_path",
+    help="Zookeeper node path from which the total size of orphaned objects will be taken.",
+)
```
Comment (Alex-Burmak) [inline ch_tools/monrun_checks/ch_orphaned_objects.py:24]: It looks a bit strange that these options start with `--store` as the command only reads the state. Suggest to rename the options to `--state-local` and `--state-zk-path` (or to something else without `--store` suffix).

**Comment by Alex-Burmak** [comment]
Is the PR in draft on purpose? Or you just forgot to click "Ready for review"?

File: ch_tools/monrun_checks/ch_orphaned_objects.py (hunk 2 above)
Comment (NikitaUnisikhin) [reply ch_tools/monrun_checks/ch_orphaned_objects.py]: thank, done

File: ch_tools/chadmin/cli/object_storage_group.py (hunk 1 above)
Comment (NikitaUnisikhin) [reply ch_tools/chadmin/cli/object_storage_group.py]: thanks, done

**Comment by NikitaUnisikhin** [comment]
Yes, I forgot to click "Ready for review"

By the way, what do you think about making the store-state-zk flag and creating a default path for storing statistics? I don’t like  that in this case there will be one path for storing statistics for hosts from different shards, but perhaps this problem should be left on the user’s side. But in some scenarios it may be useful.

**Comment by Alex-Burmak** [comment]
> By the way, what do you think about making the store-state-zk flag and creating a default path for storing statistics? I don’t like that in this case there will be one path for storing statistics for hosts from different shards, but perhaps this problem should be left on the user’s side. But in some scenarios it may be useful.

I have no strong preference between the current implementation and described rework. Choose the one you prefer.

> It would be cool in ch-tools to know which shard the host belongs to. Do we have such an opportunity? Using this the code can be made much better

Shard name can be obtained from ClickHouse config, "shard" macro. Another option is to add a setting with shard name to ch-tools config.

**Review by Alex-Burmak — APPROVED** [review]

**Comment by NikitaUnisikhin** [comment]
I'm inclined to leave the current implementation for flexibility in storing statistics in zk. 
@Alex-Burmak I don't have rights to merge. Merge please if there are no more comments on this PR.

**Pull request merged**